submissions/balancer-design/universal_8to8.md

```diff
@@ -0,0 +1,68 @@
+
+![How not to use a balancer](https://cdn.discordapp.com/attachments/699618487097884723/813449477028446269/unknown.png)
+
+In this article, I set myself a goal to make a balancer for which this sort of intuition works, in other words, no matter the input/output configuration, all outputs are always equal, and all inputs are drawn from evenly. This kind of balancer is known as a "universal balancer", a concept first explored and named in [this post](https://www.reddit.com/r/factorio/comments/a5ferf/i_present_to_you_the_44_universal_balancer/) by Reddit user [u/tzwaan](https://www.reddit.com/user/tzwaan), who is one of the moderators of the [Factorio subreddit](https://www.reddit.com/r/factorio).
+
+Universal balancers can do what no balancer should be able to: no matter the configuration, they're input-balanced, output-balanced, *and* throughput-unlimited at the same time! This is a set of properties previously thought to be possessed only by a 2-2 balancer, which is just a single splitter. The drawback is that they get *big*.
+
+### The design process
+
+Let's think about how to convert a 1-4 balancer into a 1-3. That is done by looping the extra output back into the first splitter. The same can be done with a 4-4 balancer, which we will use as our base principle:
+
+![Looped 4-4](https://cdn.discordapp.com/attachments/699618487097884723/813449600852819968/unknown.png)
+
+Now, how would we make it so the balancing doesn't get ruined when we block some outputs? Simple, we use priority splitters to auto-loop the items that can't get out. Of course, we need to merge them back with the input too, and set input priority so the main inputs don't slow down when there are items coming:
+
+![Auto-looping 4-4](https://cdn.discordapp.com/attachments/699618487097884723/813449812728217640/unknown.png)
+
+We now discover a major issue. If the looping items try to enter a belt that's already full, they just back up all the way to the split-off point and block the overflow, putting us back to square one:
+
+![Broken auto-looping 4-4](https://cdn.discordapp.com/attachments/699618487097884723/813450086234587156/unknown.png)
+
+To remedy this, we need to make sure every item on the loop can get to every single one of the loop's exits. The way to do this with the least splitters is a secondary balancer:
+
+![Fixed 4-4](https://cdn.discordapp.com/attachments/699618487097884723/813450357639872553/unknown.png)
+
+And we're pretty much done with the 4-4. It can be improved, but more on that later. Now, let's try our hand at something bigger: a universal 8-8. We do the same steps as before, and get this:
+
+![Basic universal 8-8](https://cdn.discordapp.com/attachments/699618487097884723/813450635667832842/unknown.png)
```

Comment:
A thought, is this balancer from some blueprint book or something? I think you should credit the designer of the base balancers you're using if possible.